src/Tasks/Microsoft.NET.Build.Tasks/DependencyContextBuilder.cs

```diff
@@ -125,7 +124,7 @@ public DependencyContext Build()
                 _projectContext.LockFileTarget.TargetFramework.DotNetFrameworkName,
                 _projectContext.LockFileTarget.RuntimeIdentifier,
                 runtimeSignature,
-                _projectContext.IsPortable);
+                isPortable: string.IsNullOrEmpty(_projectContext.LockFileTarget.RuntimeIdentifier));
```

Comment:
I need to add a comment here.

Basically, `IsPortable` != `IsFrameworkDependent` and `IsPortable` != `!IsSelfContained`.

But `IsFrameworkDependent` == `!IsSelfContained`.  (for now until we add some other app model where it doesn't run on a shared framework and still isn't self-contained.)

`IsPortable` means your app can run on any machine. It is portable across architectures, OSes, etc.
`IsFrameworkDependent` means your app runs on the shared framework, but it can still be runtime-specific.  note: `IsPortable` [implies](https://en.wikipedia.org/wiki/Truth_table#Logical_implication) `IsFrameworkDependent`
`IsSelfContained` means everything required to run your app is located in your publish output.

In this case, DependencyModel is strictly asking for `IsPortable`, not `IsFrameworkDependent` or `!IsSelfContained`.  And [looking under the hood](https://github.com/dotnet/core-setup/blob/master/src/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.cs#L94-L100), DependencyModel treats `IsPortable` and an empty `RuntimeIdentifier` the same.  In a perfect world, this boolean wouldn't be needed to be passed here.